### PR TITLE
(1645) Make organisation and extending_organisation consistent

### PR DESCRIPTION
**Pull request by leeky**
This data migration ensures that existing activities have `organisation` and `extending_organisation` set consistently. This mirrors the values that `ActivityDefaults` assigns for new activities depending on their level:

- Fund: organisation / extending_organisation are always both BEIS
- Programme: organisation is BEIS and extending_organisation is set to a delivery partner
- Project: organisation / extending_organisation are the same DP
- Third-party Project: organisation / extending_organisation are the same DP

In the past `extending_organisation_id` has been an optional field, so we copy this value from `organisation_id` for projects and third-party projects.

I output a warning for any programmes whose `extending_organisation` is incorrectly set to BEIS, as these will need to be manually updated, to allow the relevant delivery partner to create child activities (projects) under them.

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [ ] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

File: spec/models/activity_spec.rb [hunk 1]
```diff
@@ -77,8 +77,8 @@
       end
     end
 
-    describe ".publishable_to_iati?" do
-      it "only returns activities with a `form_state` of :complete and/or `publish_to_iati` of true" do
+    describe ".publishable_to_iati" do
+      it "only returns activities with a 'complete' and `publish_to_iati` is true" do
```
Comment (CristinaRO) [inline spec/models/activity_spec.rb:81]: While you're in there making things better (🎉 ), could you rephrase "with a 'complete'" to something else?

File: db/data/20210412134546_fix_extending_organisation_inconsistencies.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+Activity.project.update_all("extending_organisation_id = organisation_id")
+
+Activity.third_party_project.update_all("extending_organisation_id = organisation_id")
+
+bad_programmes = Activity.programme.where(extending_organisation_id: beis.id)
+
+if bad_programmes.any?
+  puts "Found #{bad_programmes.size} programme(s) whose extending_organisation is incorrectly set to BEIS and should be manually corrected:"
+  puts bad_programmes.ids.join("\n")
```
Comment (CristinaRO) [inline db/data/20210412134546_fix_extending_organisation_inconsistencies.rb:17]: How would you feel about generating the list of "bad" programmes as a list of production URLs? ("Nope" is a valid answer.)

**Review by CristinaRO — APPROVED** [review]
Approved in principle, a couple of comments.

File: db/data/20210412134546_fix_extending_organisation_inconsistencies.rb (hunk 2 above)
Comment (leeky) [reply db/data/20210412134546_fix_extending_organisation_inconsistencies.rb:17]: Nice idea, but the Activity details page is scoped on organisation, and this warning is flagging places where it is unreliable.
